### ISSUE TITLE
listsFlattened with custom column name

### ISSUE THREAD

**Issue by nWidart** [TypiCMS/NestableCollection#3]
Hello,

I'm trying to get a flattened list of a nestable collection. Though, I'm using the `name` column.
I can't seem to be able to set that column:

``` php
return $this->model->orderBy('parent_id')->get()->listsFlattened(null, 0, [], 'name');
```

This returns: `Cannot pass parameter 3 by reference`

When I remove the reference from the 3rd parameter if the `listsFlattened` method, I only get:

``` php
array:3 [▼
  1 => "En Category"
  6 => "Third Categoryyy"
  3 => "Second Category"
]
```

But even them `->listsFlattened(null, 0, [], 'name')` isn't really nice. (lot's of empty parameters).

Though, the table structure is like so:
![screen shot 2015-03-24 at 09 51 31](https://cloud.githubusercontent.com/assets/882397/6798243/6552fb06-d20b-11e4-82df-ef1e92acab23.png)

Category id 3 (`Second Category`) should be under the first category.

So to summarise:
1. It might be better to have the `$column` property as first parameter
   
   ``` php
   public function listsFlattened($column = 'title', BaseCollection $collection = null, $level = 0, array &$flattened = [], $indentChars = '&nbsp;&nbsp;&nbsp;&nbsp;')
   ```
2. Why is the item `#3` in the example above not indended ?

Thanks!

**Comment by sdebacker** [comment]
1. Why not
2. That's because you need to nest it first : `$collection->nest()->listsFlattened()`

**Comment by nWidart** [comment]
Hm, using 

``` php
return $this->model->orderBy('parent_id')->get()->nest()->listsFlattened('name');
```

I get, 

```
Argument 2 passed to TypiCMS\NestableCollection::listsFlattened() must be an instance of Illuminate\Support\Collection, integer given,
```

With the method signature changed, where `$column`, is the first parameter. The second argument is `BaseCollection $collection = null`.

Will send a PR to move the parameter if you want.

**Comment by sdebacker** [comment]
Did you change method call in line 83 ?

**Comment by sdebacker** [comment]
Also I see on line 83 that two parameters are missing…

**Comment by sdebacker** [comment]
Ok for the pull request.

**Comment by nWidart** [comment]
Oh, indeed. Forgot to change the recursive call. 

Pull request incoming. :smile:
